statistics/statistics.go

```diff
@@ -60,14 +60,94 @@ type Column struct {
 	//
 	// Repeat is the number of repeats of the bucket value, it can be used to find popular values.
 	//
-	// We could have make a bucket struct contains number, value and repeat, but that would be harder to
-	// serialize, so we store every bucket field in its own slice instead. those slices all have
-	// the bucket count length.
+	// TODO: We could have make a bucket struct contains number, value and repeats.
 	Numbers []int64
 	Values  []types.Datum
 	Repeats []int64
 }
 
+func (c *Column) saveToStorage(ctx context.Context, tableID int64, isIndex bool) error {
+	var colName string
+	if isIndex {
+		colName = "index_id"
+	} else {
+		colName = "col_id"
+	}
+	insertSQL := fmt.Sprintf("insert into mysql.stats_columns (table_id, %s, distinct_count) values (%d, %d, %d)", colName, tableID, c.ID, c.NDV)
+	_, err := ctx.(sqlexec.SQLExecutor).Execute(insertSQL)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	for i := 0; i < len(c.Numbers); i++ {
+		var count int64
+		if i == 0 {
+			count = c.Numbers[i]
+		} else {
+			count = c.Numbers[i] - c.Numbers[i-1]
+		}
+		val, err := c.Values[i].ConvertTo(ctx.GetSessionVars().StmtCtx, types.NewFieldType(mysql.TypeBlob))
+		if err != nil {
+			return errors.Trace(err)
+		}
+		if isIndex {
+			insertSQL = fmt.Sprintf("insert into mysql.stats_buckets values(%d, NULL, %d, %d, %d, %d, X'%X')", tableID, c.ID, i, count, c.Repeats[i], val.GetBytes())
+		} else {
+			insertSQL = fmt.Sprintf("insert into mysql.stats_buckets values(%d, %d, NULL, %d, %d, %d, X'%X')", tableID, c.ID, i, count, c.Repeats[i], val.GetBytes())
+		}
+		_, err = ctx.(sqlexec.SQLExecutor).Execute(insertSQL)
+		if err != nil {
+			return errors.Trace(err)
+		}
+	}
+	return nil
+}
+
+func colStatsFromStorage(ctx context.Context, tableID int64, colID int64, tp *types.FieldType, distinct int64, isIndex bool) (*Column, error) {
```

Comment:
We should use different structs to represent column stats and index stats.
index stats may contain multiple columns.